initial commit: hw1

Oleg-Nik opened the pull request.

Oleg-Nik committed:
initial commit

Oleg-Nik committed:
initial commit
for test

Oleg-Nik committed:
comments added

a-chernov commented on hw01.py:
```diff
@@ -0,0 +1,30 @@
+sorted_list = list(range(list_length)) # create dummy for sorted list
+list_of_random_numbers = list(range(list_length)) # create dummy for random list
+# create list with random number
+for _ in range(list_length): list_of_random_numbers[_] = random.randint(1, probability_range)
```
https://www.python.org/dev/peps/pep-0008/#other-recommendations

Oleg-Nik replied:
done

a-chernov commented on hw01.py:
```diff
@@ -0,0 +1,30 @@
+list_length = 100
+probability_range = 1000 # define Length of the list and the Depth of probability
+sorted_list = list(range(list_length)) # create dummy for sorted list
+list_of_random_numbers = list(range(list_length)) # create dummy for random list
+# create list with random number
+for _ in range(list_length): list_of_random_numbers[_] = random.randint(1, probability_range)
```
Ты делаешь лишнее действие, зачем занимать память лишним объектом - списком из 6ой строки.
Создай пустой список и в цикле добавляй в него новые случайные значения.

Oleg-Nik replied:
cycles chahged

a-chernov commented on hw01.py:
```diff
@@ -0,0 +1,30 @@
+# sorting the List
+for i in range(list_length):
+    min_number = min(list_of_random_numbers)
+    sorted_list[i] = min_number
```
То же самое здесь

Oleg-Nik replied:
done

a-chernov commented on hw01.py:
```diff
@@ -0,0 +1,30 @@
+print("------->sorted random List:")
+print(sorted_list)
+# Sums calculation
+even_sum = 0; odd_sum = 0
```
Снова https://www.python.org/dev/peps/pep-0008/#other-recommendations

Oleg-Nik replied:
changed

a-chernov commented on hw01.py:
```diff
@@ -0,0 +1,30 @@
+for i in range(list_length):
+    if sorted_list[i] % 2 == 1:
+        odd_sum = odd_sum + sorted_list[i]
+print("The sum of even numbers:", even_sum)
+print("The sum of odd numbers:", odd_sum)
```
Думаю, в задании просили среднее, а не сумму

Oleg-Nik committed:
1. Averages instead Sums.
2. list_of_random_numbers = [] --- using empty list leads to error in line 11 "IndexError: list assignment index out of range"

Oleg-Nik committed:
Delete hw02.py

Oleg-Nik committed:
I don't understand why this out-of-index error occur.
And another WHY this is working when lists are defined and not empty

Oleg-Nik committed:
Merge remote-tracking branch 'origin/hw1' into hw1

# Conflicts:
#	hw02.py

Oleg-Nik committed:
Delete hw02.py

Oleg-Nik committed:
list-comprehension added, prints changed

Oleg-Nik commented:
corrected
